### PR TITLE
Feature/sentry implementation

### PR DESCRIPTION
**Pull request by jamesstottmoj**
This PR adds sentry-sdk as a dependency and initialises the SDK ready for use with the AP-UI

### CODE REVIEW

File: ap/settings/common.py [hunk 1]
```diff
@@ -81,6 +82,32 @@
...
 ]
 
+# -- Sentry error tracking
+
+if os.environ.get("SENTRY_DSN"):
+    SENTRY_ENVIRONMENT = ENV
+    KUBERNETES_ENV = "EKS"
+    # May need to change the below if
+    if ENV == "alpha":
+        SENTRY_ENVIRONMENT = "prod"
```
Comment (jacobwoffenden) [inline ap/settings/common.py:92]: I don't think we need this logic anymore, that is control panel

File: requirements.txt [hunk 2]
```diff
@@ -15,4 +15,5 @@ pytest==8.0.0
 pytest-django==4.8.0
 python-dotenv==1.0.1
 requests==2.32.3
+sentry-sdk==1.40.5
```
Comment (jacobwoffenden) [inline requirements.txt:18]: `2.9.0` is available

File: ap/settings/common.py [hunk 3]
```diff
@@ -81,6 +82,29 @@
     "django_structlog.middlewares.RequestMiddleware",
 ]
 
+# -- Sentry error tracking
+
+if os.environ.get("SENTRY_DSN"):
+    SENTRY_ENVIRONMENT = ENV
+    KUBERNETES_ENV = "EKS"
```
Comment (michaeljcollinsuk) [inline ap/settings/common.py:89]: Not sure if we really need this? Presume its from control panel back from when it wasnt on EKS

Comment (michaeljcollinsuk) [reply ap/settings/common.py:89]: @jacobwoffenden what do you think?

File: ap/settings/common.py [hunk 4]
```diff
@@ -81,6 +82,29 @@
     "django_structlog.middlewares.RequestMiddleware",
 ]
 
+# -- Sentry error tracking
+
+if os.environ.get("SENTRY_DSN"):
+    SENTRY_ENVIRONMENT = ENV
```
Comment (michaeljcollinsuk) [inline ap/settings/common.py:88]: Not sure there is any point to this either, just use `ENV` below?

**Review by michaeljcollinsuk — CHANGES_REQUESTED** [review]
Thanks for adding this. Couple of minor suggestions

File: ap/settings/common.py (hunk 3 above)
Comment (jacobwoffenden) [reply ap/settings/common.py:89]: Agreed

File: ap/settings/common.py (hunk 4 above)
Comment (jamesstottmoj) [reply ap/settings/common.py]: Sorted

File: ap/settings/common.py [hunk 5]
```diff
@@ -81,6 +82,27 @@
...
+
+    sentry_sdk.init(
+        dsn=os.environ["SENTRY_DSN"],
+        environment=ENV,
+        integrations=[DjangoIntegration()],
+        traces_sample_rate=0.0,
+        send_default_pii=True,
+    )
+
+    if "runworker" in sys.argv:
```
Comment (jamesstottmoj) [inline ap/settings/common.py:101]: Do we need this block? Unsure how useful this is

Comment (jamesstottmoj) [reply ap/settings/common.py:101]: From line 101 down

Comment (michaeljcollinsuk) [reply ap/settings/common.py:101]: Agreed, lets remove it

**Review by michaeljcollinsuk — APPROVED** [review]
LGTM

**Pull request merged**